packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts

```diff
@@ -190,27 +193,63 @@ export const getByViewerHandler = async ({ ctx }: GetByViewerOptions) => {
     },
   });
 
+  const teamMemberships = user.teams.map((membership) => ({
+    teamId: membership.team.id,
+    membershipRole: membership.role,
+  }));
+
+  const compareMembership = (mship1: MembershipRole, mship2: MembershipRole) => {
+    const mshipToNumber = (mship: MembershipRole) =>
+      Object.keys(MembershipRole).findIndex((mmship) => mmship === mship);
+    return mshipToNumber(mship1) > mshipToNumber(mship2);
+  };
+
   eventTypeGroups = ([] as EventTypeGroup[]).concat(
     eventTypeGroups,
-    user.teams.map((membership) => ({
-      teamId: membership.team.id,
-      membershipRole: membership.role,
-      profile: {
-        name: membership.team.name,
-        image: `${CAL_URL}/team/${membership.team.slug}/avatar.png`,
-        slug: membership.team.slug ? "team/" + membership.team.slug : null,
-      },
-      metadata: {
-        membershipCount: membership.team.members.length,
-        readOnly: membership.role === MembershipRole.MEMBER,
-      },
-      eventTypes: membership.team.eventTypes
-        .map(mapEventType)
-        .filter((evType) => evType.userId === null || evType.userId === ctx.user.id)
-        .filter((evType) =>
-          membership.role === MembershipRole.MEMBER ? evType.schedulingType !== SchedulingType.MANAGED : true
-        ),
-    }))
+    user.teams
+      .filter((mmship) => {
+        const metadata = teamMetadataSchema.parse(mmship.team.metadata);
+        return !metadata?.isOrganization;
+      })
+      .map((membership) => {
+        const orgMembership = teamMemberships.find(
+          (teamM) => teamM.teamId === membership.team.parentId
+        )?.membershipRole;
+        return {
+          teamId: membership.team.id,
+          membershipRole:
+            orgMembership && compareMembership(orgMembership, membership.role)
+              ? orgMembership
+              : membership.role,
+          profile: {
+            name: membership.team.name,
+            image: `${CAL_URL}/team/${membership.team.slug}/avatar.png`,
+            slug: membership.team.slug
+              ? !membership.team.parentId
+                ? `/team`
+                : "" + membership.team.slug
+              : null,
+          },
+          metadata: {
+            membershipCount: membership.team.members.length,
+            readOnly:
+              membership.role ===
+              (membership.team.parentId
+                ? orgMembership && compareMembership(orgMembership, membership.role)
+                  ? orgMembership
+                  : MembershipRole.MEMBER
+                : MembershipRole.MEMBER),
```

Comment:
In order to define if an event-type is read-only, we need to see the org membership for the user, because in case the user is an admin or owner of the org, same membership should apply to each sub-team